pycbc/waveform/bank.py

```diff
@@ -661,17 +661,23 @@ def get_decompressed_waveform(self, tempout, index, f_lower=None,
         if self.waveform_decompression_method is not None :
             decompression_method = self.waveform_decompression_method
         else :
-            decompression_method = self.compressed_waveforms[self.table.template_hash[index]].interpolation
+            decompression_method = self.compressed_waveforms[self.table[index].template_hash].interpolation
```

Comment:
If we're going really seriously down the optimization route. It might be quicker to avoid making this call three times. As:
```
self.table[index]
```
is called multiple times, maybe do:
```
table_idx = self.table[index]
temp_hash = table_idx.template_hash
```
and then use those in lines 671, 675 and 664.